### PR TITLE
Add onboarding-scripts package

### PR DESCRIPTION
**Pull request by mar3kk**
Signed-off-by: Marek Kubiczek <marek.kubiczek@imgtec.com>

### CODE REVIEW

File: onboarding-scripts/Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=onboarding-scripts
+#PKG_VERSION:=dev
+PKG_VERSION:=1.0.0
+PKG_RELEASE:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=$(TOPDIR)/../../packages/onboarding-scripts
```
Comment (nikhil-zinjurde-imgtec) [inline onboarding-scripts/Makefile]: The source URL should be remote i.e. https://github.com/CreatorDev/ci40-onboarding-scripts.git

Developers should set the source link locally for development purpose

**Review by nikhil-zinjurde-imgtec — CHANGES_REQUESTED** [review]

File: onboarding-scripts/Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+endef
+
+define Build/Compile
+		$(call Build/Compile/Default,all)
+endef
+
+define Build/Clean
+		$(call Build/Compile/Default,clean)
+endef
+
```
Comment (ghost) [inline onboarding-scripts/Makefile]: Do we actually need to define Build/Compile and Build/Clean if you are using the default functions ?

File: onboarding-scripts/Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+define Package/onboarding-scripts/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_DIR) $(1)/etc/avahi/services
+	$(INSTALL_DIR) $(1)/usr/lib/lua
+	$(INSTALL_DIR) $(1)/www
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/* $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/etc/init.d/* $(1)/etc/init.d/
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/etc/avahi/services/* $(1)/etc/avahi/services/
+	cp -r $(PKG_INSTALL_DIR)/usr/lib/lua/* $(1)/usr/lib/lua/
```
Comment (ghost) [inline onboarding-scripts/Makefile]: $(CP)

File: onboarding-scripts/Makefile [hunk 4]
```diff
@@ -0,0 +1,55 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=onboarding-scripts
+#PKG_VERSION:=dev
```
Comment (ghost) [inline onboarding-scripts/Makefile]: remove comment

File: onboarding-scripts/Makefile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+PKG_RELEASE:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=$(TOPDIR)/../../packages/onboarding-scripts
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_VERSION:=$(PKG_VERSION)
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+
```
Comment (ghost) [inline onboarding-scripts/Makefile:13]: You are missing PKG_MAINTAINER, PKG_LICENSE, PKG_LICENSE_FILE.

File: onboarding-scripts/Makefile [hunk 6]
```diff
@@ -0,0 +1,55 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=onboarding-scripts
+#PKG_VERSION:=dev
+PKG_VERSION:=1.0.0
+PKG_RELEASE:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (ghost) [inline onboarding-scripts/Makefile]: PKG_BUILD_DIR is defined by default as: $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION).

File: onboarding-scripts/Makefile (hunk 2 above)
Comment (Ham22) [reply onboarding-scripts/Makefile]: Nope

File: onboarding-scripts/Makefile (hunk 5 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile:13]: What Do You think should I put in PKG_MAINTAINER? I do not see these fields defined in any of other packages.

Comment (Ham22) [reply onboarding-scripts/Makefile:13]: For now we don't, we will worry about this if we ever get around to upstreaming it

File: onboarding-scripts/Makefile (hunk 1 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile]: done

Comment (mar3kk) [reply onboarding-scripts/Makefile]: done

File: onboarding-scripts/Makefile [hunk 7]
```diff
@@ -0,0 +1,49 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=onboarding-scripts
+PKG_VERSION:=v0.1.0
+PKG_RELEASE:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (ghost) [inline onboarding-scripts/Makefile]: No need to redefine PKG_BUILD_DIR. Remove it

File: onboarding-scripts/Makefile [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_VERSION:=$(PKG_VERSION)
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+
+PKG_LICENSE_FILE:=LICENSE
+PKG_LICENSE:=BSD-3-Clause
+
+CMAKE_INSTALL:=1
+
+CMAKE_OPTIONS += -DSTAGING_DIR=$(STAGING_DIR)
```
Comment (ghost) [inline onboarding-scripts/Makefile]: Why do you need to set this option ?

File: onboarding-scripts/Makefile [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+define Package/onboarding-scripts/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_DIR) $(1)/etc/avahi/services
+	$(INSTALL_DIR) $(1)/usr/lib/lua
+	$(INSTALL_DIR) $(1)/www
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/* $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/etc/init.d/* $(1)/etc/init.d/
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/etc/avahi/services/* $(1)/etc/avahi/services/
+	$(CP) -r $(PKG_INSTALL_DIR)/usr/lib/lua/* $(1)/usr/lib/lua/
```
Comment (ghost) [inline onboarding-scripts/Makefile]: no need for the -r option.
It is already part of the definition of CP in rules.mk

File: onboarding-scripts/Makefile [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+define Package/onboarding-scripts/description
+  Imagination Technologies Ci40 onboarding scripts
+endef
+
+define Package/onboarding-scripts/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_DIR) $(1)/etc/avahi/services
+	$(INSTALL_DIR) $(1)/usr/lib/lua
+	$(INSTALL_DIR) $(1)/www
```
Comment (ghost) [inline onboarding-scripts/Makefile:46]: Could you group INSTALL_DIR and CP ? So you install www, binaries,libraries...

File: onboarding-scripts/Makefile [hunk 11]
```diff
@@ -0,0 +1,49 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=onboarding-scripts
+PKG_VERSION:=v0.1.0
```
Comment (ghost) [inline onboarding-scripts/Makefile]: It seems that very few packages have PKG_VERSION defined as v[number]. I suggest
```
PKG_VERSION:=0.1.0
...
PKG_SOURCE_VERSION:=v$(PKG_VERSION)
```

File: onboarding-scripts/Makefile (hunk 8 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile]: I believe this is copied from other project and not needed. Removed.

File: onboarding-scripts/Makefile (hunk 9 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile]: fixed

File: onboarding-scripts/Makefile (hunk 10 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile:46]: grouped

File: onboarding-scripts/Makefile (hunk 11 above)
Comment (mar3kk) [reply onboarding-scripts/Makefile]: I decided to go with pure version number for tags as well. fixed.

**Review by ghost — APPROVED** [review]

**Review by nikhil-zinjurde-imgtec — APPROVED** [review]

**Pull request merged**